stories/s2l2a.stories.js

```diff
@@ -341,6 +343,59 @@ export const GetMapProcessingEvalscripturlVersion2 = () => {
   return wrapperEl;
 };
 
+export const GetMapProcessingAlternativeHostname = () => {
+  if (!process.env.CLIENT_ID || !process.env.CLIENT_SECRET) {
+    return "<div>Please set OAuth Client's id and secret for Processing API (CLIENT_ID, CLIENT_SECRET env vars)</div>";
+  }
+  if (!process.env.ALTERNATIVE_HOSTNAME) {
+    return '<div>Please set ALTERNATIVE_HOSTNAME to use this story</div>';
+  }
+
+  const img = document.createElement('img');
+  img.width = '512';
+  img.height = '512';
+
+  const wrapperEl = document.createElement('div');
+  wrapperEl.innerHTML = `<h2>GetMap - on alternative hostname (${process.env.ALTERNATIVE_HOSTNAME})</h2>`;
+  wrapperEl.insertAdjacentElement('beforeend', img);
+
+  const perform = async () => {
+    registerHostnameReplacing('services.sentinel-hub.com', process.env.ALTERNATIVE_HOSTNAME);
+
+    try {
+      // await setAuthTokenWithOAuthCredentials();
+      const authToken = await requestAuthToken(process.env.CLIENT_ID, process.env.CLIENT_SECRET);
+      setAuthToken(authToken);
+      console.log('Auth token retrieved and set successfully');
+
+      const layerS2L2A = new S2L2ALayer({
+        instanceId,
+        layerId,
+        evalscript: 'return [2.5 * B04, 2.5 * B03, 2.5 * B02];',
+      });
+
+      const getMapParams = {
+        bbox: bbox4326,
+        fromTime: new Date(Date.UTC(2018, 11 - 1, 22, 0, 0, 0)),
+        toTime: new Date(Date.UTC(2018, 12 - 1, 22, 23, 59, 59)),
+        width: 512,
+        height: 512,
+        format: MimeTypes.JPEG,
+      };
+      const imageBlob = await layerS2L2A.getMap(getMapParams, ApiType.PROCESSING);
+      img.src = URL.createObjectURL(imageBlob);
+    } finally {
+      // reset auth token back, so that we don't break other stories:
+      registerHostnameReplacing('services.sentinel-hub.com', 'services.sentinel-hub.com');
+      const authTokenServices = await requestAuthToken(clientId, clientSecret);
```

Comment:
`clientId` and `clientSecret` are not defined. I think `process.env.CLIENT_ID` and `process.env.CLIENT_SECRET`  can be used instead.

Reply:
Was fixed. Resolving